runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java

```diff
@@ -127,5 +169,135 @@ private void setDodgyCharges(int dodgyCharges)
 	{
 		this.dodgyCharges = dodgyCharges;
 		config.dodgyNecklace(dodgyCharges);
+		if (currentInfoBoxIDs.get(EquipmentInventorySlot.AMULET) == ItemID.DODGY_NECKLACE)
+		{
+			if (dodgyCharges > 0)
+			{
+				createJewelleryInfobox(ItemID.DODGY_NECKLACE, dodgyCharges, EquipmentInventorySlot.AMULET);
+			}
+			else
+			{
+				removeJewelleryInfobox(EquipmentInventorySlot.AMULET);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(ItemContainerChanged event)
+	{
+		if (event.getItemContainer().getItems().length > 14)
+		{
+			return;
+		}
+
+		Item[] items = event.getItemContainer().getItems();
+
+		int amulet = -2;
+		int bracelet = -2;
+		int ring = -2;
+		if (items.length >= 3)
+		{
+			if (config.showAmuletInfobox())
+			{
+				amulet = items[EquipmentInventorySlot.AMULET.getSlotIdx()].getId();
+			}
+			if (items.length >= 10)
+			{
+				if (config.showBraceletInfobox())
+				{
+					bracelet = items[EquipmentInventorySlot.GLOVES.getSlotIdx()].getId();
+				}
+				if (items.length >= 13)
+				{
+					if (config.showRingInfobox())
+					{
+						ring = items[EquipmentInventorySlot.RING.getSlotIdx()].getId();
+					}
+				}
+			}
+		}
+		else
+		{
+			return;
+		}
+
+		if (amulet > 0)
```

Comment:
Not sure if this would be desired but it reduces blocks

```
		ItemWithCharge amuletCharges = ItemWithCharge.findItem(amulet);
		if (amuletCharges != null || (amulet == ItemID.DODGY_NECKLACE && dodgyCharges > 0))
		{
			createJewelleryInfobox(amulet, (amulet == ItemID.DODGY_NECKLACE ? dodgyCharges : amuletCharges.getCharges()), EquipmentInventorySlot.AMULET);
		}
		else
		{
			removeJewelleryInfobox(EquipmentInventorySlot.AMULET);
		}
```

Reply:
The idea with this layout is it makes the code easy to understand for the outliers. There are a handful of items that store charges like the Dodgy Necklace, and I don't think anyone wants a single if statement with 14 different conditions, or 6 nested ternaries.

Reply:
Then you would have 1 variable instead of 14 different variables for each because you can only wear 1

Reply:
What do you mean 1 variable? Each item has to track its charges separately for the overlay anyway?

Reply:
You are wanting overlay for worn items?

Reply:
The overlay that already exists in this plugin that shows the charges on top of the item
![image](https://user-images.githubusercontent.com/2979691/50152580-e07dee00-02bb-11e9-8ee0-2c27dc718704.png)

Reply:
you are wanting an infobox for each one or just worn?

Reply:
Just the one that's worn, but that doesn't change that the charge overlay needs to track every single one of them at once, so there needs to be a store for each one.